v2/spanner-change-streams-to-sharded-file-sink/src/test/java/com/google/cloud/teleport/v2/templates/SpannerChangeStreamToGcsMultiShardIT.java

```diff
@@ -219,4 +223,81 @@ private void assertFileContentsInGCSForMultipleShards() {
     assertThatArtifacts(artifactsShardC).hasContent("SingerId\\\":\\\"4");
     assertThatArtifacts(artifactsShardC).hasContent("SingerId\\\":\\\"5");
   }
+
+  @Test
+  public void testForwardMigrationFiltered() throws IOException, java.lang.InterruptedException {
+    // Construct a ChainedConditionCheck with below stages.
+    // 1. Wait for the metadata table to have the start time of reader job
+    // 2. Write 1 records per shard to Spanner with the transaction tag as txBy=
+    // 3. Wait and check there are no files in GCS for that shard
+    ChainedConditionCheck conditionCheck =
+        ChainedConditionCheck.builder(
+                List.of(
+                    SpannerRowsCheck.builder(
+                            spannerMetadataResourceManager, "spanner_to_gcs_metadata")
+                        .setMinRows(1)
+                        .setMaxRows(1)
+                        .build()))
+            .build();
+    // Wait for conditions
+    PipelineOperator.Result result =
+        pipelineOperator()
+            .waitForCondition(createConfig(jobInfo, Duration.ofMinutes(10)), conditionCheck);
+    // Assert Conditions
+    assertThatResult(result).meetsConditions();
+    // Perform writes to Spanner
+    writeSpannerDataForForwardMigration(7, "seven", "testShardD");
+    // Assert file present in GCS with the needed data
+    assertFileContentsInGCSForFilteredRecords();
+  }
+
+  private void assertFileContentsInGCSForFilteredRecords() {
+    ChainedConditionCheck conditionCheck =
+        ChainedConditionCheck.builder(
+                List.of(
+                    GCSArtifactsCheck.builder(
+                            gcsResourceManager, "output/testShardD/", Pattern.compile(".*\\.txt$"))
+                        .setMinSize(0)
+                        .setMaxSize(0)
```

Comment:
How does this work? Won't it immediately match since no file = 0 records?

Reply:
Thanks, updated the assert